elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLDimension.java

```diff
@@ -5,51 +5,59 @@
  */
 package com.yahoo.elide.datastores.aggregation.queryengines.sql.metadata;
 
-import static com.yahoo.elide.datastores.aggregation.queryengines.sql.SQLQueryEngine.generateColumnReference;
 import static com.yahoo.elide.datastores.aggregation.queryengines.sql.SQLQueryEngine.getClassAlias;
 
 import com.yahoo.elide.core.EntityDictionary;
 import com.yahoo.elide.core.Path;
 import com.yahoo.elide.datastores.aggregation.core.JoinPath;
+import com.yahoo.elide.datastores.aggregation.metadata.LabelResolver;
+import com.yahoo.elide.datastores.aggregation.metadata.LabelStore;
+import com.yahoo.elide.datastores.aggregation.metadata.models.Column;
 import com.yahoo.elide.datastores.aggregation.metadata.models.Dimension;
 import com.yahoo.elide.datastores.aggregation.metadata.models.Table;
-import com.yahoo.elide.datastores.aggregation.queryengines.sql.annotation.JoinTo;
 
 import lombok.Getter;
 
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * SQLDimension are dimension columns with extra physical information.
  */
 public class SQLDimension extends Dimension implements SQLColumn {
     @Getter
-    private final String reference;
-
-    @Getter
-    private final JoinPath joinPath;
+    private String reference;
 
     @Getter
-    private EntityDictionary metadataDictionary;
+    private List<JoinPath> joinPaths = new ArrayList<>();
 
     public SQLDimension(Table table, String fieldName, EntityDictionary dictionary) {
         super(table, fieldName, dictionary);
-        Class<?> tableClass = dictionary.getEntityClass(table.getId());
+    }
+
+    @Override
+    public Path getSourcePath(EntityDictionary metadataDictionary) {
```

Comment:
I believe sourcePath and sourceColumn are going away - based on our last discussion.